cmd/query/app/server_test.go

```diff
@@ -53,10 +62,314 @@ func TestCreateTLSServerError(t *testing.T) {
+		// {
+		// 	name: "",
+		// 	serverTLS: tlscfg.Options{
+		// 		Enabled:      true,
+		// 		CertPath:     "invalid/path",
+		// 		KeyPath:      "invalid/path",
+		// 		ClientCAPath: "invalid/path",
+		// 	},
+		// 	clientTLS: tlscfg.Options{
+		// 		Enabled:    true,
+		// 		CertPath:   "invalid/path",
+		// 		KeyPath:    "invalid/path",
+		// 		CAPath:     "invalid/path",
+		// 		ServerName: "example.com",
+		// 	},
+		// 	expectError:      false,
+		// 	expectServerFail: false,
+		// },
+		{
+			name: "Should pass with insecure connection",
+			serverTLS: tlscfg.Options{
+				Enabled: false,
+			},
+			clientTLS: tlscfg.Options{
+				Enabled: false,
+			},
+			expectError:      false,
+			expectServerFail: false,
+		},
+		{
+			name: "should fail with TLS client to untrusted TLS server",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				ServerName: "example.com",
+			},
+			expectError:      true,
+			expectServerFail: false,
+		},
+		{
+			name: "should fail with TLS client to trusted TLS server with incorrect hostname",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "nonEmpty",
+			},
+			expectError:      true,
+			expectServerFail: false,
+		},
+		{
+			name: "should pass with TLS client to trusted TLS server with correct hostname",
+			serverTLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "example.com",
+			},
+			expectError:      false,
+			expectServerFail: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			serverOptions := &QueryOptions{TLSHTTP: test.serverTLS}
+			httpServer, err := createHTTPServer(&querysvc.QueryService{}, serverOptions, opentracing.NoopTracer{}, zap.NewNop())
+
+			if test.expectServerFail {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.NotNil(t, httpServer)
+			}
+			httpListener, err := net.Listen("tcp", "localhost:"+fmt.Sprintf("%d", ports.QueryHTTP))
+			if test.expectServerFail {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			if serverOptions.TLSHTTP.Enabled {
+
+				tlsCfg, _ := serverOptions.TLSHTTP.Config()
+				tlsCfg.Rand = rand.Reader
+				tlsHTTPListener := tls.NewListener(httpListener, tlsCfg)
+				go func() {
+					// err
+					_ = httpServer.Serve(tlsHTTPListener)
```

Comment:
let's try to assert no error, with exception of ServerClosed

Reply:
actually i did that before.  but both this subroutine and the outer subroutine both use `t *testing.T`  which lead to race conditions.  Hence i ignored it.  the best we could do is log the message.  Assertion wouldnt work.